### PR TITLE
Reduce "Implicit conversion" warnings

### PR DESCRIPTION
**Pull request by PokeCodec**
Modern compilers tend to complain about this implicit casting, so I added an explicit cast in these areas.

### CODE REVIEW

**Comment by PikalaxALT** [comment]
Converting this to draft until all instances of this issue are resolved.

**Comment by PokeCodec** [comment]
I made a LOT of refactorings so that we could minimize these casting warnings while still matching.

It took a while, but I did it.

File: src/battle_anim_effects_1.c [hunk 1]
```diff
@@ -4445,7 +4445,7 @@ static void AnimBowMon_Step1_Callback(struct Sprite* sprite)
     {
         sprite->data[3] = gBattlerSpriteIds[gBattleAnimAttacker];
         PrepareBattlerSpriteForRotScale(sprite->data[3], ST_OAM_OBJ_NORMAL);
-        sprite->data[4] = (sprite->data[6] = GetBattlerSide(gBattleAnimAttacker)) ? 0x300 : 0xFFFFFD00;
+        sprite->data[4] = (sprite->data[6] = GetBattlerSide(gBattleAnimAttacker)) ? 0x300 : (s16)0xFFFFFD00;
```
Comment (DizzyEggg) [inline src/battle_anim_effects_1.c:4448]: 0xFFFFFD00 is just -0x300, no casts needed

**Comment by PokeCodec** [comment]
Fine, I cut back to those that make the code more complicated to read.

File: src/secret_base.c [hunk 2]
```diff
@@ -964,7 +964,7 @@ static void HandleRegistryMenuInput(u8 taskId)
 
     data = gTasks[taskId].data;
     input = ListMenu_ProcessInput(data[5]);
-    ListMenuGetScrollAndRow(data[5], &data[2], &data[1]);
+    ListMenuGetScrollAndRow((u8)data[5], (u16 *)&data[2], (u16 *)&data[1]);
```
Comment (PikalaxALT) [inline src/secret_base.c:967]: Explicit cast from a signed integer type to a narrower unsigned integer type is not required.

**Comment by PokeCodec** [comment]
Done!

**Comment by PokeCodec** [comment]
@PikalaxALT

**Pull request merged**